models/risk_models.py

```diff
@@ -45,38 +51,46 @@ def update_risk_local(cls, human, now):
         return 0.0
 
     @classmethod
-    def add_message_to_cluster(cls, human, m_i):
-        """ This function clusters new messages by scoring them against old messages in a sort of naive nearest neighbors approach"""
-        # TODO: include risk level in clustering, currently only uses quantized uid
-        # TODO: refactor to compare multiple clustering schemes
-        # TODO: check for mutually exclusive messages in order to break up a group and re-run nearest neighbors
-        m_i_enc = _encode_message(m_i)
-        m_risk = binary_to_float("".join([str(x) for x in np.array(m_i[1].tolist()).astype(int)]), 0, 4)
-
-        # otherwise score against previous messages
+    def score_matches(cls, human, m_i):
         scores = {}
-        for m_enc, _ in human.M.items():
-            m = _decode_message(m_enc)
-            if m_i[0] == m[0] and m_i[2].day == m[2].day:
+        for m_enc, _ in reversed(human.M.items()):
```

Comment:
It seams `__reverse__` on `dict_items` was only introduced in Python 3.8. Consider using `for m_enc, _ in reversed(list(human.M.items()))` for backward compatibility to at least Python 3.6

Reply:
good call -- updated.